roscala/src/test/scala/coop/rchain/rosette/CtxtSpec.scala

```diff
@@ -0,0 +1,59 @@
+package coop.rchain.rosette
+
+import org.scalatest.{FlatSpec, Matchers}
+import coop.rchain.rosette.Location._
+import coop.rchain.rosette.Ob.Lenses._
+
+class CtxtSpec extends FlatSpec with Matchers {
+  val ctxt = Ctxt.empty
+    .copy(id = 0)
+    .copy(ctxt = Ctxt.empty.copy(id = 1))
+
+  "applyK" should "save result into parent ctxt" in {
+    val ctxt0 = Ctxt.applyK(Fixnum(1), LocTrgt).runS(ctxt).value
+
+    ctxt0.ctxt.trgt should be(Fixnum(1))
+  }
+
+  it should "return parent ctxt as a continuation if outstanding field of parent is one" in {
+    val ctxt0  = ctxt.set(_ >> 'ctxt >> 'outstanding)(1)
+    val result = Ctxt.applyK(Fixnum(1), LocTrgt).runA(ctxt0).value
+
+    result._2.isDefined should be(true)
```

Comment:
I think we can provide additional information to a reader using destructuring in assigning:
```scala
val (first,second) = Ctxt.applyK(Fixnum(1), LocTrgt).runA(ctxt0).value
second.isDefined shoudBe true
```

Reply:
True.